### DISCUSSION TITLE
NowCast vs AQI Current Conditions

### DISCUSSION THREAD

**Discussion by dshanske** [bellrichm/weewx-aqi-xtype#4] [General]
If I'm reading the technical documentation currently, there are two EPA AQI methods.

AQI is an averaged amount over a 24 hour period for pm2.5 and pm10. So, once the day is over, this is the one that would be used for the calendar day.

The NowCast is computed from the most recent 12 hours, but weighing the most recent hours of data more heavily than an ordinary 12-hour average. It is used in lieu of a 24-hour average PM concentration in the calculation of the AQI until an entire calendar day is completed.

Curious what your plans are, because averaging out is a hard thing to do.

**Comment by bellrichm** [comment]
At the moment I was trying to wrap my head around NowCast. Your explanation helps a lot! Agree on the complexity. 

Currently I don’t have a need for NowCast, so no plans. But that could easily change. If I did do something, seems like a ‘current’ value would be what makes sense….

I’d like to hear your thoughts on how you would use it.

**Reply by dshanske** [reply]
Well, nowcast makes sense for current conditions, but not for statistics such as the day or historically.

Also, current probably shouldn't be a recommendation in the readme... probably should be the hourly because it moves so much in an hour. I refresh every 180 seconds.

**Comment by bellrichm** [comment]
After spending more time researching this, I am going to take a crack at it. Configuring it would look something like this.
```
[aqitype]
    # The name of AQI field.
    # Create a section for each field to be calculated.
    [[pm2_5_aqi_nowcast]]
        # The name of the WeeWX observation to be used in the calculation.
        input = pm2_5
        # The name of the algorithm.
        algorithm = NOWCAST
        # If the algorithm supports different pollutants(pm 2.5, pm 10, etc)
        # Supported values: pm2_5, pm10
        type = pm2_5       
```
It would only be available from $current and $last.
